src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -11,42 +11,75 @@ import { localize } from "../../localize";
 export async function filterDownloadAppSettings(context: IActionContext, sourceSettings: { [key: string]: string }, destinationSettings: { [key: string]: string }, destinationName: string): Promise<void> {
     let suppressPrompt: boolean = false;
     let overwriteSetting: boolean = false;
+    let suppressPromptOfSettingsToIgnore = false;
+    let overwriteSettingsToIgnore: boolean = false;
```

Comment:
are the `suppressPromptOfSettingsToIgnore` and `overwriteSettingsToIgnore` in the context of each setting? if so, they should be declared within the for loop and not outside it. Right now, reading the code make it confusing on whether they are function properties or within the context of each setting.

Reply:
The way the original flow had it set up was that if at any time the user suppresses the overwriting prompts it would apply to all app settings and not just in the context of each setting. Should I follow that logic or have it be that suppressing the prompt only applies to the current setting at hand?

Reply:
so I think someone reading the code here will be confused that you have `suppressPrompt` and `overwriteSetting` at the function level along with `showPromptForsettingstoIgnore` and `overwriteSettingsToIgnore`. Basically it is hard to understand why there are a two function level variables serving the same purpose.. also the latter two are only used within the context of the loop. 

I dont fully understand your question, maybe we can discuss offline.

Reply:
sure, I'll talk to you about it today at the sync.